homeassistant/components/airgradient/button.py

```diff
@@ -0,0 +1,79 @@
+"""Support for AirGradient buttons."""
+
+from collections.abc import Awaitable, Callable
+from dataclasses import dataclass
+
+from airgradient import AirGradientClient, ConfigurationControl
+
+from homeassistant.components.button import ButtonEntity, ButtonEntityDescription
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ServiceValidationError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import DOMAIN, AirGradientConfigEntry
+from .coordinator import AirGradientConfigCoordinator
+from .entity import AirGradientEntity
+
+
+@dataclass(frozen=True, kw_only=True)
+class AirGradientButtonEntityDescription(ButtonEntityDescription):
+    """Describes AirGradient button entity."""
+
+    press_fn: Callable[[AirGradientClient], Awaitable[None]]
+
+
+CO2_CALIBRATION = AirGradientButtonEntityDescription(
+    key="co2_calibration",
+    translation_key="co2_calibration",
+    press_fn=lambda client: client.request_co2_calibration(),
+)
+LED_BAR_TEST = AirGradientButtonEntityDescription(
+    key="led_bar_test",
+    translation_key="led_bar_test",
+    press_fn=lambda client: client.request_led_bar_test(),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: AirGradientConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up AirGradient button entities based on a config entry."""
+    model = entry.runtime_data.measurement.data.model
+    coordinator = entry.runtime_data.config
+
+    entities = [AirGradientButton(coordinator, CO2_CALIBRATION)]
+    if "L" in model:
+        entities.append(AirGradientButton(coordinator, LED_BAR_TEST))
+
+    async_add_entities(entities)
+
+
+class AirGradientButton(AirGradientEntity, ButtonEntity):
+    """Defines an AirGradient button."""
+
+    entity_description: AirGradientButtonEntityDescription
+    coordinator: AirGradientConfigCoordinator
+
+    def __init__(
+        self,
+        coordinator: AirGradientConfigCoordinator,
+        description: AirGradientButtonEntityDescription,
+    ) -> None:
+        """Initialize airgradient button."""
+        super().__init__(coordinator)
+        self.entity_description = description
+        self._attr_unique_id = f"{coordinator.serial_number}-{description.key}"
+
+    async def async_press(self) -> None:
+        """Press the button."""
+        if (
+            self.coordinator.data.configuration_control
+            is not ConfigurationControl.LOCAL
```

Comment:
Can we mark them as unavailable when the configuration_control is cloud?
Or maybe even better, we should not add these entities if configuration_control is cloud

Reply:
But that would mean we have to create and delete entities on the fly if the config source is set, i don't think that's a nice user experience

Reply:
Then we should mark them as unavailable imo